integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -416,6 +416,31 @@ object CommonUtil {
     }
   }
 
+  /**
+   * This method will validate the cache expiration time specified by the user
+   *
+   * @param tableProperties table property specified by user
+   * @param propertyName property name
+   */
+  def validateCacheExpiration(tableProperties: Map[String, String], propertyName: String): Unit = {
+    var expirationTime: java.lang.Long = 0L
+    if (tableProperties.get(propertyName).isDefined) {
+      val value = tableProperties(propertyName)
+      val exceptionMsg = s"Invalid $propertyName value found: " +
+                         s"$value, only duration from 1 minute to 1440 minutes is supported."
+      try {
+        expirationTime = java.lang.Long.parseLong(value)
+      } catch {
+        case n: NumberFormatException =>
+          throw new MalformedCarbonCommandException(exceptionMsg)
+      }
+      if (expirationTime == 0L) {
+        throw new MalformedCarbonCommandException(exceptionMsg)
+      }
+      tableProperties.put(propertyName, value)
```

Comment:
As it is a segment level expiry (because key for cache is mergeindex file name), better to allow alter table properties to change value for new loads.

Reply:
It is currently added as a table-level segment expiry, means, expiration is for table level cache. Alter table properties is still needed?

Reply:
done